website/docs/language/modules/develop/refactoring.html.md

```diff
@@ -0,0 +1,475 @@
+
+  # ...
+}
+
+module "y" {
+  source = "../modules/y"
+
+  # ...
+}
+
+moved {
+  from = aws_instance.a
+  to   = module.x.aws_instance.a
+}
+
+moved {
+  from = aws_instance.b
+  to   = module.x.aws_instance.b
+}
+
+moved {
+  from = aws_instance.c
+  to   = module.y.aws_instance.c
+}
+```
+
+When an existing user of the original module upgrades to the new "shim"
+version, Terraform will notice these three `moved` blocks and thus behave
+as if the objects associated with the three old resource addresses were
+originally created inside the two new modules.
+
+New users of this family of modules may use either the combined shim module
+_or_ the two new modules separately. You may wish to communicate to your
+existing users that the old module is now deprecated and so they should use
+the two separate modules for any new needs.
+
+The multi-module refactoring situation is unusual in that it violates the
+typical rule that a parent module sees its child module as a "closed box",
+unaware of exactly which resources are declared inside it. This compromise
+assumes that all three of these modules are maintained by the same people
+and distributed together in a single
+[module package](sources.html#modules-in-package-sub-directories).
+
+To avoid strong [coupling](https://en.wikipedia.org/wiki/Coupling_(computer_programming))
+between separately-packaged modules, Terraform will only allow declarations of
+moves between modules in the same package. In other words, Terraform would
+not have allowed moving into `module.x` above if the `source` address of
+that call had not been a [local path](sources.html#local-paths).
+
+Module references in `moved` blocks are resolved relative to the module
+instance they are defined in. For example, if our original module above were
+already a child module named `module.original`, the reference to
+`module.x.aws_instance.a` would resolve as
+`module.original.module.x.aws_instance.a`. A module may only make `moved`
+statements about its own objects and objects of its child modules.
+
+If you need to refer to resources within a module that was called using
+`count` or `for_each` meta-arguments, you must specify a specific instance
+key to use in order to match with the new location of the resource
+configuration:
+
+```hcl
+moved {
+  from = aws_instance.example
+  to   = module.new[2].aws_instance.example
+}
+```
+
+## Removing `moved` Blocks
+
+Over time, a long-lasting module may accumulate many `moved` blocks.
+
+If you are maintaining private modules within an organization and you know
+for certain that all users of your module have successfully run
+`terraform apply` with your new module version already then it can be safe
+to remove `moved` blocks in later versions of your module.
+
+However, in the general case removing a `moved` block is a breaking change
+to your module, because any configurations whose state refers to the old
+address will then plan to delete that existing object instead of move it.
```

Comment:
```suggestion
However, removing a `moved` block is a generally breaking change
to your module because any configurations whose state refers to the old
address will then plan to delete that existing object instead of move it.
```